compiler/src/test/java/com/squareup/anvil/compiler/dagger/InjectConstructorFactoryGeneratorTest.kt

```diff
@@ -675,6 +675,22 @@ public final class InjectClass_Factory implements Factory<InjectClass> {
     }
   }
 
+  @Test fun `a factory class is generated for an inject constructor with a lambda as super type`() {
+    compile(
+      """
+        package com.squareup.test
+
+        import javax.inject.Inject
+  
+        class InjectClass @Inject constructor() : (Int) -> Unit {
```

Comment:
You didn't want to write a test for the 23-parameter version? ;)

Reply:
No, thanks :-P